### ISSUE TITLE
Runtime metrics charts broken

### ISSUE THREAD

**Issue by jmsmkn** [comic/grand-challenge.org#3470]
For an algorithm job.

<img width="1148" alt="Screenshot 2024-08-05 at 12 44 18" src="https://github.com/user-attachments/assets/158ce48c-6c0a-40d8-93e6-64eee9995aa8">

The data are there:

```
<div class="w-100 vega-lite-chart">
                        <script id="runtimeMetricsData" type="application/json">{"$schema": "https://vega.github.io/schema/vega-lite/v5.json", "width": "container", "padding": 0, "title": "ml.g4dn.xlarge / 4 CPU / 16 GB Memory / 1x T4 GPU", "data": {"values": [{"Metric": "CPUUtilization", "Timestamp": "2024-06-13T13:41:00+00:00", "Percent": 1.00498}, {"Metric": "CPUUtilization", "Timestamp": "2024-06-13T13:40:00+00:00", "Percent": 1.0066899999999999}, {"Metric": "CPUUtilization", "Timestamp": "2024-06-13T13:39:00+00:00", "Percent": 1.0058799999999999}, {"Metric": "CPUUtilization", "Timestamp": "2024-06-13T13:38:00+00:00", "Percent": 1.00556}, {"Metric": "CPUUtilization", "Timestamp": "2024-06-13T13:37:00+00:00", "Percent": 1.00546}, {"Metric": "CPUUtilization", "Timestamp": "2024-06-13T13:36:00+00:00", "Percent": 0.92246}, {"Metric": "CPUUtilization", "Timestamp": "2024-06-13T13:35:00+00:00", "Percent": 0.897324}, {"Metric": "CPUUtilization", "Timestamp": "2024-06-13T13:34:00+00:00", "Percent": 0.6173879999999999}, {"Metric": "DiskUtilization", "Timestamp": "2024-06-13T13:41:00+00:00", "Percent": 0.000638435}, {"Metric": "DiskUtilization", "Timestamp": "2024-06-13T13:40:00+00:00", "Percent": 0.0006380139999999999}, {"Metric": "DiskUtilization", "Timestamp": "2024-06-13T13:39:00+00:00", "Percent": 0.000637592}, {"Metric": "DiskUtilization", "Timestamp": "2024-06-13T13:38:00+00:00", "Percent": 0.000637171}, {"Metric": "DiskUtilization", "Timestamp": "2024-06-13T13:37:00+00:00", "Percent": 0.000636749}, {"Metric": "DiskUtilization", "Timestamp": "2024-06-13T13:36:00+00:00", "Percent": 0.000636327}, {"Metric": "DiskUtilization", "Timestamp": "2024-06-13T13:35:00+00:00", "Percent": 0.0006358}, {"Metric": "DiskUtilization", "Timestamp": "2024-06-13T13:34:00+00:00", "Percent": 0.000635624}, {"Metric": "GPUMemoryUtilization", "Timestamp": "2024-06-13T13:41:00+00:00", "Percent": 0.320117}, {"Metric": "GPUMemoryUtilization", "Timestamp": "2024-06-13T13:40:00+00:00", "Percent": 0.319987}, {"Metric": "GPUMemoryUtilization", "Timestamp": "2024-06-13T13:39:00+00:00", "Percent": 0.319987}, {"Metric": "GPUMemoryUtilization", "Timestamp": "2024-06-13T13:38:00+00:00", "Percent": 0.319987}, {"Metric": "GPUMemoryUtilization", "Timestamp": "2024-06-13T13:37:00+00:00", "Percent": 0.308268}, {"Metric": "GPUMemoryUtilization", "Timestamp": "2024-06-13T13:36:00+00:00", "Percent": 0.286003}, {"Metric": "GPUMemoryUtilization", "Timestamp": "2024-06-13T13:35:00+00:00", "Percent": 0.0}, {"Metric": "GPUMemoryUtilization", "Timestamp": "2024-06-13T13:34:00+00:00", "Percent": 0.0}, {"Metric": "GPUUtilization", "Timestamp": "2024-06-13T13:41:00+00:00", "Percent": 1.0}, {"Metric": "GPUUtilization", "Timestamp": "2024-06-13T13:40:00+00:00", "Percent": 0.92}, {"Metric": "GPUUtilization", "Timestamp": "2024-06-13T13:39:00+00:00", "Percent": 0.7}, {"Metric": "GPUUtilization", "Timestamp": "2024-06-13T13:38:00+00:00", "Percent": 0.7}, {"Metric": "GPUUtilization", "Timestamp": "2024-06-13T13:37:00+00:00", "Percent": 0.7}, {"Metric": "GPUUtilization", "Timestamp": "2024-06-13T13:36:00+00:00", "Percent": 1.0}, {"Metric": "GPUUtilization", "Timestamp": "2024-06-13T13:35:00+00:00", "Percent": 0.0}, {"Metric": "GPUUtilization", "Timestamp": "2024-06-13T13:34:00+00:00", "Percent": 0.0}, {"Metric": "MemoryUtilization", "Timestamp": "2024-06-13T13:41:00+00:00", "Percent": 0.23245100000000002}, {"Metric": "MemoryUtilization", "Timestamp": "2024-06-13T13:40:00+00:00", "Percent": 0.229462}, {"Metric": "MemoryUtilization", "Timestamp": "2024-06-13T13:39:00+00:00", "Percent": 0.22946000000000003}, {"Metric": "MemoryUtilization", "Timestamp": "2024-06-13T13:38:00+00:00", "Percent": 0.225717}, {"Metric": "MemoryUtilization", "Timestamp": "2024-06-13T13:37:00+00:00", "Percent": 0.22197399999999998}, {"Metric": "MemoryUtilization", "Timestamp": "2024-06-13T13:36:00+00:00", "Percent": 0.217}, {"Metric": "MemoryUtilization", "Timestamp": "2024-06-13T13:35:00+00:00", "Percent": 0.027233200000000003}, {"Metric": "MemoryUtilization", "Timestamp": "2024-06-13T13:34:00+00:00", "Percent": 0.00886259}]}, "layer": [{"transform": [{"calculate": "100*datum.Percent", "as": "Percent100"}], "encoding": {"x": {"timeUnit": "hoursminutesseconds", "field": "Timestamp", "title": "Local Time / HH:MM:SS"}, "y": {"field": "Percent100", "type": "quantitative", "title": "Utilization / %"}, "color": {"field": "Metric", "type": "nominal"}}, "layer": [{"mark": "line"}, {"transform": [{"filter": {"param": "hover", "empty": false}}], "mark": "point"}]}, {"transform": [{"pivot": "Metric", "value": "Percent", "groupby": ["Timestamp"]}], "mark": "rule", "encoding": {"opacity": {"condition": {"value": 0.3, "param": "hover", "empty": false}, "value": 0}, "tooltip": [{"field": "CPUUtilization", "type": "quantitative", "format": ".2%"}, {"field": "DiskUtilization", "type": "quantitative", "format": ".2%"}, {"field": "GPUMemoryUtilization", "type": "quantitative", "format": ".2%"}, {"field": "GPUUtilization", "type": "quantitative", "format": ".2%"}, {"field": "MemoryUtilization", "type": "quantitative", "format": ".2%"}], "x": {"timeUnit": "hoursminutesseconds", "field": "Timestamp", "title": "Local Time / HH:MM:SS"}}, "params": [{"name": "hover", "select": {"type": "point", "fields": ["Timestamp"], "nearest": true, "on": "mouseover", "clear": "mouseout"}}]}, {"data": {"values": [{}]}, "mark": {"type": "rule", "strokeDash": [8, 8]}, "encoding": {"y": {"datum": 400}}}, {"data": {"values": [{}]}, "mark": {"type": "text", "baseline": "line-bottom"}, "encoding": {"text": {"datum": "CPU Utilization Limit"}, "y": {"datum": 400}}}, {"data": {"values": [{}]}, "mark": {"type": "rule", "strokeDash": [8, 8]}, "encoding": {"y": {"datum": 100}}}, {"data": {"values": [{}]}, "mark": {"type": "text", "baseline": "line-bottom"}, "encoding": {"text": {"datum": "Memory / GPU / GPU Memory Utilization Limit"}, "y": {"datum": 100}}}]}</script>
                    </div>
```

Likely caused by #3460

**Comment by jmsmkn** [comment]
@ammar257ammar fixing this should be your top priority

**Comment by ammar257ammar** [comment]
Indeed, it is related to the mentioned issue. The problem is that not all the pages with charts having the class vega-lite-chart uses the javascript file refresh_sidebar.mjs (in which the event trigger was added in the issue fix).
I reverted to the previous state and added a different fix for #3460. I will make a PR.

On my local dev, I don't have the "Runtime Metrics" section in the job results. How I can obtain that to test the fix?

**Comment by jmsmkn** [comment]
You can hardcode it, I put the contents of the block in the issue.

**Comment by ammar257ammar** [comment]
Thanks!! I can confirm it works. I will make a PR

**Comment by jmsmkn** [comment]
Fixed by reverting #3460, that needs to be re-implemented in a way that does not break other things.
